### PR TITLE
Moving all dependencies to explicit Libs constants

### PR DESCRIPTION
**Pull request by AHeise**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -17,5 +17,5 @@
 ################################################################################
 org.gradle.caching=true
 org.gradle.parallel=true
-org.gradle.configureondemand=true
+#org.gradle.configureondemand=true
```
Comment (aljoscha) [inline gradle.properties]: What's the reason for this?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.5.1-bin.zip
```
Comment (aljoscha) [inline gradle/wrapper/gradle-wrapper.properties]: Why the downgrade to 5.5.1?

File: buildSrc/src/main/kotlin/Versions.kt [hunk 3]
```diff
@@ -19,13 +19,13 @@ object Versions {
 
   const val aliyun_sdk_oss: String = "3.4.1" // available: "3.6.0"
 
-  const val aws_java_sdk_core: String = "1.11.271" // available: "1.11.615"
+  const val aws_java_sdk_core: String = "1.11.271" // available: "1.11.618"
```
Comment (aljoscha) [inline buildSrc/src/main/kotlin/Versions.kt]: There are some version changes in here. Are they on purpose?

File: flink-mesos/build.gradle.kts [hunk 4]
```diff
@@ -3,21 +3,21 @@ plugins {
...
-    implementation("org.apache.mesos:mesos:1.0.1")
-    implementation("com.netflix.fenzo:fenzo-core:0.10.1")
-    implementation("org.apache.flink:flink-shaded-netty:4.1.32.Final-7.0")
-    testImplementation("org.scalatest:scalatest_${Versions.baseScala}:3.0.0")
-    testImplementation("org.apache.curator:curator-test:2.12.0")
-    testImplementation("com.typesafe.akka:akka-testkit_${Versions.baseScala}:2.5.21")
+    implementation(Libs.mesos)
+    implementation(Libs.fenzo_core)
+    implementation(Libs.flink_shaded_netty)
+    testImplementation(Libs.scalatest_2_11)
```
Comment (aljoscha) [inline flink-mesos/build.gradle.kts]: I think we still need to do some cleanups, for example use `Versions.baseScala` in the Scala dependencies so we don't have the explicit `2_11` here.

File: gradle.properties (hunk 1 above)
Comment (AHeise) [reply gradle.properties]: I had some issues with JDK version not being applied to subprojects, so I disabled this beta feature.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (AHeise) [reply gradle/wrapper/gradle-wrapper.properties]: No particular reason, you just forgot to add the wrapper and I created a new one with my version.
I will revert to 5.6.

File: buildSrc/src/main/kotlin/Versions.kt (hunk 3 above)
Comment (AHeise) [reply buildSrc/src/main/kotlin/Versions.kt]: I reran the srcVersions plugin and it added the newest version in the comment.

File: flink-mesos/build.gradle.kts (hunk 4 above)
Comment (AHeise) [reply flink-mesos/build.gradle.kts]: Yes, but much later. First get it to compile. Then make it more beautiful.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (rmetzger) [reply gradle/wrapper/gradle-wrapper.properties]: Does this actually work?
```
➜  flink-gradle git:(aheise-gradle-poc) ✗ ./gradlew build
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
```

File: buildSrc/src/main/kotlin/Versions.kt (hunk 3 above)
Comment (aljoscha) [reply buildSrc/src/main/kotlin/Versions.kt]: I think we should stick to the versions that Flink currently has.

File: flink-mesos/build.gradle.kts (hunk 4 above)
Comment (aljoscha) [reply flink-mesos/build.gradle.kts]: 👌

**Pull request merged**